src/app/backend/file-parser/path-parser.spec.ts

```
@@ -130,4 +164,224 @@ describe('PathParser', () => {
+        );
+    });
+
+    it('should throw PointOfInterested error (Type missing)', () => {
+        template[7] = '<fake></fake>\n';
+        const path = new Path([], []);
+        const parser = new PathParser(path);
+        expect(() => { parser.parse(template.join('')); }).toThrow(new ParserError(
+            'PointOfInterest entry number 0 of Segment of Id 1 must have \"Type\" as 1st tag.')
+        );
+    });
+
+    it('should throw PointOfInterested error (Name missing)', () => {
+        template[8] = '<fake></fake>\n';
+        const path = new Path([], []);
+        const parser = new PathParser(path);
+        expect(() => { parser.parse(template.join('')); }).toThrow(new ParserError(
+            'PointOfInterest entry number 0 of Segment of Id 1 must have \"Name\" as 2nd tag.')
+        );
+    });
+
+    it('should throw PointOfInterested error (Position missing)', () => {
+        template[9] = '<fake></fake>\n';
+        const path = new Path([], []);
+        const parser = new PathParser(path);
+        expect(() => { parser.parse(template.join('')); }).toThrow(new ParserError(
+            'PointOfInterest entry number 0 of Segment of Id 1 must have \"Position\" as 3rd tag.')
+        );
+    });
+
+    it('should throw error if Segment and SegmentAppearance entries do not appear in the same order', () => {
+        template[15] = '<Id>16</Id>\n';
+        const path = new Path([], []);
+        const parser = new PathParser(path);
+        expect(() => { parser.parse(template.join('')); }).toThrow(new ParserError(
+            'Missmatch of Id\'s / ordering between SegmentAppearance and Segment.\n' +
+            'SegmentAppearance of Id 16 was matched with Segment of Id 1.')
+        );
+    });
+
+    it ('should throw error if there are multiple coordinate definitions for a POI', () => {
+        template.splice(12, 0, '    <Segment>\n');
```

Same as other comment, you edit a lot of the xml data here, but it is unclear what happens and why this should throw an error.